Domains: Make the "free" flag display consistent

sixhours opened the pull request:
* Swap green for dark gray
* Remove text-transform and bold font
* Add left margin

**Before this PR**

<img width="1058" alt="screen shot 2018-08-13 at 10 51 29 am" src="https://user-images.githubusercontent.com/2124984/44039431-0cc1f17a-9ee7-11e8-81f5-84f7c1bba457.png">

**After this PR**

<img width="1058" alt="screen shot 2018-08-13 at 10 50 24 am" src="https://user-images.githubusercontent.com/2124984/44039430-0cb490ca-9ee7-11e8-8ae8-d7a007efbadb.png">

**Steps to test**

1) Switch to this PR and navigate to `/domains/add/`
2) Ensure the active site has a paid plan with a domain credit that has not been used
3) Check the "Free with your plan" text for the above changes

klimeryk commented:
I've tested this and it looks good when there's enough space, but when the viewport is narrower, I think there's extra margin that should not be there:
<img width="610" alt="screen shot 2018-08-22 at 16 10 33" src="https://user-images.githubusercontent.com/3392497/44472553-4f0cb100-a626-11e8-947d-343ced5586eb.png">

Previously:
<img width="591" alt="screen shot 2018-08-22 at 16 13 56" src="https://user-images.githubusercontent.com/3392497/44472600-677ccb80-a626-11e8-963b-d90767e5573e.png">

sixhours commented:
@klimeryk Oh, good catch! Will take a look and patch that. Thanks.

klimeryk commented:
Sure thing. One more I was just editing in the comment (but I'll put it in a separate one so that you won't miss it) - if the text is in one column like that, all three lines look very similar. Meaning, similar size and color - it's kind of an existing issue with the design, but the green of the previous design (and the existing lighter gray of the stroke out price) made it easier to differentiate between the lines. Maybe we could improve it somehow? Making the domain suggestion itself bold seems like an obvious one, but maybe there's something else you'd recommend.

sixhours commented:
I've fixed the display issues on small screens and updated the font size/color a bit to take the edge off all the gray:

<img width="1060" alt="screen shot 2018-08-22 at 2 15 17 pm" src="https://user-images.githubusercontent.com/2124984/44482282-e4537980-a615-11e8-8bc0-d25475a04794.png">

<img width="507" alt="screen shot 2018-08-22 at 2 15 28 pm" src="https://user-images.githubusercontent.com/2124984/44482283-e4537980-a615-11e8-9e2d-bc94446cb4c3.png">

I want to avoid bolding the domain title since we don't use that pattern elsewhere in domain search results. I've also added a bit of breathing room between the lines so they don't run together and hopefully increase readability. We could also go a bit lighter on the struck-out price opacity to reduce the contrast and bring the "free" part forward more. @jancavan @klimeryk Let me know what you think!

klimeryk commented:
Oooh, you're gonna hate this PR 😅 I like the improvements regarding the colors - I agree that making the domain name bold would introduce inconsistencies.

So I got to thorough testing and noticed two more issues (sorry, should have noticed them earlier probably):
It seems that at a certain breakpoint, the `Select` button jumps to its own row:

<img width="477" alt="screen shot 2018-08-30 at 00 53 20" src="https://user-images.githubusercontent.com/3392497/44820203-22d9cd00-abf0-11e8-8299-9224511c0e3b.png">

<img width="433" alt="screen shot 2018-08-30 at 00 53 30" src="https://user-images.githubusercontent.com/3392497/44820208-25d4bd80-abf0-11e8-895b-9a5f85cdefc1.png">

And issue 2: if you have a DWPO user (a user that can only buy domains with a plan) _and_ they have domain credit available (so they bought a plan, but don't have a domain bundled with it), we don't show the stroke out price (because those users can't buy their first domains a'la carte, so it doesn't make sense to show those prices), but it seems the new design does not take this into account:
<img width="959" alt="screen shot 2018-08-30 at 00 58 52" src="https://user-images.githubusercontent.com/3392497/44820343-bdd2a700-abf0-11e8-876c-7fbf668b0fde.png">

The old design looked like this:
<img width="948" alt="screen shot 2018-08-30 at 01 07 10" src="https://user-images.githubusercontent.com/3392497/44820451-418c9380-abf1-11e8-897e-4b600d2a535c.png">

They're nothing major, but since we're improving this I figured you wanted to do it once and proper :) Oh, and just to be clear - I've tested all these flows for different PRs countless times, so I know all the edge cases and where regressions usually creep in 😁 But I think those two are the only ones left 🙇 

Feel free to ping me on Slack if you have problems reproducing the second issue (I have a test user you can use).

jancavan commented:
![screenshot 2018-08-30 10 29 00](https://user-images.githubusercontent.com/4924246/44868382-86d7b200-ac3f-11e8-9a91-ea07628e1089.png)

In `/domains/add/`, we also need to remove the indentation before "Free with your plan."

jancavan commented:
Sorry, I didn't realize @klimeryk was referring to the same issue.

So to reproduce:
1. Make sure your site has a plan (mine has a Business plan)
2. From the sidebar, go to Domains (`/domains/add/`)
3. Search for a domain to add

sixhours commented:
@klimeryk The Select button jumps to its own row on purpose for small screens; that was changed in #25789 to better accommodate longer domain names and languages that lengthen the select button.

> They're nothing major, but since we're improving this I figured you wanted to do it once and proper :) Oh, and just to be clear - I've tested all these flows for different PRs countless times, so I know all the edge cases and where regressions usually creep in

I'm always worried I'm missing things, so thank you for flagging this! It should be fixed now. :)
